### PR TITLE
Add server version to output

### LINKED ISSUE #7: Display server version
The version of the server (if reported by the server) has been added to the federation tester's output in the following form:

```
  "Version": {
    "name": "Synapse",
    "version": "0.99.1.1"
  }
```

at the top level. Would be good to display in the UI.

### PR DESCRIPTION
**Pull request by anoadragon453**
Display server version information in output.

Closes #7

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -112,6 +113,21 @@ let App = create({
...
         <TestResults json={this.state.json}/>
       </>
+
+      // Display server version information
+      const serverVersion = this.state.json.Version;
+      let versionText;
+      if (serverVersion == undefined) {
+        versionText = "This homeserver does not supply version information"
+      } else if (serverVersion.error != undefined) {
+        versionText = serverVersion.error;
```
Comment (babolivier) [inline app.js]: I'm really not sure this is the right way to do this. When testing, it gives me

```
Version information: Get matrix://fake-labs.abolivier.bzh/_matrix/federation/v1/version: dial tcp: lookup fake-labs.abolivier.bzh on 1.1.1.1:53: no such host
```

which isn't really the most explicit way to put it imho. Replacing `Version information: [...]` with `An error happened when looking up the homeserver's version: [...]` when there's an error sounds sensible to me.

File: app.js [hunk 2]
```diff
@@ -112,6 +113,21 @@ let App = create({
...
+      } else {
+        versionText = serverVersion.name + " " + serverVersion.version;
+      }
+      
+      version = <>
+        <p>Version information: {versionText}</p>
```
Comment (babolivier) [inline app.js]:
```suggestion
        <p>Homeserver version: {versionText}</p>
```

Might be better to keep that part specific to the "we got a version and no error" case in the `if...else if...else` blocks above.

File: app.js [hunk 3]
```diff
@@ -112,6 +113,21 @@ let App = create({
...
+
+      // Display server version information
+      const serverVersion = this.state.json.Version;
+      let versionText;
+      if (serverVersion == undefined) {
+        versionText = "This homeserver does not supply version information"
+      } else if (serverVersion.error != undefined) {
+        versionText = serverVersion.error;
+      } else {
+        versionText = serverVersion.name + " " + serverVersion.version;
```
Comment (babolivier) [inline app.js]:
```suggestion
        versionText = `${serverVersion.name} ${serverVersion.version}`;
```

File: app.js [hunk 4]
```diff
@@ -137,6 +153,7 @@ let App = create({
             <div className="sk-cube sk-cube9"></div>
           </div>
         </div>
+        {version}
```
Comment (babolivier) [inline app.js]: I'd move that to `result`, which could look like:

```jsx
      result = <>
        Got {reportCount} connection report{reportCount > 1 && <>s</>}.
        {reportCount == 0 && <> This usually means at least one error happened.</>}

        {reportCount > 0 &&
          <div className="serverVersion">
            {version}
          </div>
        }

        <div className="tldr">
          {this.state.tldr}
        </div>
        <TestResults json={this.state.json}/>
      </>
```

**Comment by anoadragon453** [comment]
Great feedback, thanks!

File: app.js [hunk 5]
```diff
@@ -102,11 +103,28 @@ let App = create({
     }
 
     if (this.state.json != undefined) {
+      // Display server version information
+      const serverVersion = this.state.json.Version;
+      let version;
+      if (serverVersion == undefined) {
+        version = "This homeserver does not supply version information"
+      } else if (serverVersion.error != undefined) {
+        version = "There was an error looking up homserver version information: serverVersion.error";
```
Comment (babolivier) [inline app.js]:
```suggestion
        version = `There was an error looking up homserver version information: ${serverVersion.error}`;
```

Comment (anoadragon453) [reply app.js]: Oh lord, how did I manage that. Thanks for spotting.

**Pull request merged**